zeebe/docs/rolling_updates.md

```diff
@@ -1,5 +1,76 @@
 # Rolling Updates
 
+Starting with 8.5.0, Zeebe officially supports rolling updates.
+
+## Breaking Changes
+
+To support rolling updates, we have to ensure that all patch version of the previous or current minor are compatible with the development version.
+
+### Network
+
+Old and new versions need to be able to communicate with each other.
+Most importantly, they need to be able to form a raft replication group, and update cluster membership via SWIM.
+The command API should also remain compatible between old and new versions but exceptions can be made.
+Breaking command API compatibility drastically increases the impact of rolling updates because it may block all processing until the rolling update is finished.
+
+Compatibility can be achieved through differently, depending on the use case.
+For example:
+1. New version supports both old and new protocol and can automatically choose which to use.
+2. New version supports a new protocol but doesn't use it by default.
+3. New version detects an old version and has explicit update logic.
+
+The first option is the classic solution to evolve network protocols.
+To make this possible, we recommend to always version protocols, for example by prefixing all messages with a single integer version.
+This makes it possible for old versions to detect newer protocol versions and for new versions to support both protocol versions at the same time.
+See `AbstractMessageDecoder` and its implementations for an (old) example.
+Details are documented in our [messaging](./messaging.md#protocol-format) docs.
+
+The second option should be avoided if possible because it requires more effort.
+When switching to a new protocol requires a configuration change, this has a higher impact than the rolling update itself.
+Additionally, the old protocol has to be deprecated and can only be removed later.
+When removing a deprecated protocol, there is the risk that unaware users encounter problems during the rolling update.
+
+The third option is the one we chose when implementing dynamic scaling where new versions would communicate via a new gossip protocol that old versions did not support.
+The new version avoids the need to communicate with an old version because it can assume the state of the old version and make decisions without communicating.
+See `RollingUpdateAwareInitializerV83ToV84` for implementation details.
+
+### Processing
+
+Processing is ideally compatible between old and new versions.
+This is not always possible, for example because the new version supports a new feature or has changed the behavior of an existing feature.
+In that case we must fail early and stay on the safe side by stopping processing.
```

Comment:
❌ Processing is exactly the place to introduce changes like new features or change behavior. 

On processing, we can easily write new events (either wholly new or a different version) for new features or different behavior like bug fixes. 

🔧 I think what is meant here is what happens when an older broker encounters a record (command/event) that it is unfamiliar with (either a completely unknown record or an unknown record version). And that the older broker should fail early by stopping replay/processing in that case.

EDIT: Reading further, it becomes clear that the above meant to say this. I feel like these three lines need further clarification. I suggest changing it slightly to clarify that this discusses record incompatibility between older and newer brokers. And adding the distinction that developers are supposed to fix bugs (and change behavior) by changing how commands are processed, not by changing how events are applied.